iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParserBaseVisitorTest.java

```diff
@@ -0,0 +1,211 @@
+    var code = """
+      # example comment
+      foo=bar
+      #comment at the end""";
+
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile # example comment\n" +
+        "foo=bar\n" +
+        "#comment at the end<EOF>",
+      "visitRow # example comment\n",
+      "visitComment # example comment\n",
+      "visitEol \n",
+      "visitRow foo=bar\n",
+      "visitLine foo=bar\n",
+      "visitKey foo",
+      "visitKey bar",
+      "visitEol \n",
+      "visitRow #comment at the end<EOF>",
+      "visitComment #comment at the end<EOF>",
+      "visitEol <EOF>");
+  }
+
+  @Test
+  void shouldParseURI() {
+    var code = "spring.activemq.broker-url=tcp://127.0.0.1:61616";
+
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile spring.activemq.broker-url=tcp://127.0.0.1:61616<EOF>",
+      "visitRow spring.activemq.broker-url=tcp://127.0.0.1:61616<EOF>",
+      "visitLine spring.activemq.broker-url=tcp://127.0.0.1:61616<EOF>",
+      "visitKey spring.activemq.broker-url",
+      "visitKey tcp://127.0.0.1:61616",
+      "visitEol <EOF>");
+  }
+
+  @Test
+  void shouldParseLoggingPattern() {
+    var code = "logging.pattern.console=%d{mm:ss.SSS} %-5p [%-31t] [%-54logger{0}] %marker%m%ex{full} - %logger - %F:%L%n";
+
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile logging.pattern.console=%d{mm:ss.SSS} %-5p [%-31t] [%-54logger{0}] %marker%m%ex{full} - %logger - %F:%L%n<EOF>",
+      "visitRow logging.pattern.console=%d{mm:ss.SSS} %-5p [%-31t] [%-54logger{0}] %marker%m%ex{full} - %logger - %F:%L%n<EOF>",
+      "visitLine logging.pattern.console=%d{mm:ss.SSS} %-5p [%-31t] [%-54logger{0}] %marker%m%ex{full} - %logger - %F:%L%n<EOF>",
+      "visitKey logging.pattern.console",
+      "visitKey %d{mm:ss.SSS} %-5p [%-31t] [%-54logger{0}] %marker%m%ex{full} - %logger - %F:%L%n",
+      "visitEol <EOF>");
+  }
+
+  @Test
+  void shouldParseListProperties() {
+    var code = """
+      mail.defaultRecipients[0]=admin@mail.com
+      mail.defaultRecipients[1]=owner@mail.com""";
+
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile mail.defaultRecipients[0]=admin@mail.com\n" +
+        "mail.defaultRecipients[1]=owner@mail.com<EOF>",
+      "visitRow mail.defaultRecipients[0]=admin@mail.com\n",
+      "visitLine mail.defaultRecipients[0]=admin@mail.com\n",
+      "visitKey mail.defaultRecipients[0]",
+      "visitKey admin@mail.com",
+      "visitEol \n",
+      "visitRow mail.defaultRecipients[1]=owner@mail.com<EOF>",
+      "visitLine mail.defaultRecipients[1]=owner@mail.com<EOF>",
+      "visitKey mail.defaultRecipients[1]",
+      "visitKey owner@mail.com",
+      "visitEol <EOF>");
+  }
+
+  @Test
+  void shouldParseDifferentNewLines() {
+    var code = "foo1=bar1\nfoo2=bar2\rfoo3=bar3\r\nfoo4=4\n";
```

Comment:
I would also suggest adding tests with empty lines between properties to see how many trailing newlines are included in the text. And generally, I would suggest adding more examples based on the repo with the grammar (multiline values, etc.)

Reply:
Specifically this file contains a lot of cases: 
https://github.com/antlr/grammars-v4/blob/master/properties/examples/example1.txt